packages/amino/src/encoding.ts

```diff
@@ -77,6 +81,84 @@ export function decodeBech32Pubkey(bechEncoded: string): Pubkey {
   return decodeAminoPubkey(data);
 }
 
+/**
+ * Uvarint decoder for Amino.
+ * Decoding numbers > 127 is not supported here. Please tell those lazy CosmJS maintainers to port the binary.Varint implementation from the Go standard library and write some tests.
+ *
+ * @see https://github.com/tendermint/go-amino/blob/8e779b71f40d175/decoder.go#L64-76
+ * @returns varint as number, and bytes count occupied by varaint
+ */
+function decodeUvarint(reader: number[]): [number, number] {
+  if (reader.length < 1) {
+    throw new Error("Can't decode varint. EOF");
+  }
+  return [reader[0], 1];
+}
+
+/**
+ * Decodes a multisig pubkey to type object.
+ * Pubkey structure [ prefix + const + threshold + loop:(const + pubkeyLength + pubkey            ) ]
+ *                  [   4b   + 1b    +  varint   + loop:(1b    +    varint    + pubkeyLength bytes) ]
+ * @param reader encoded pubkey
+ */
+function decodeMultisigPubkey(reader: number[]): MultisigThresholdPubkey {
```

Comment:
Let's keep the input a `data: Uint8Array` here in the function signature. Then do `const reader = Array.from(data)` internally. Otherwiese we would have to check that every `number` is an int between 0 and 255 and nothing crazy like negative, float or NaN.

Reply:
great note